anomalib/models/cflow/model.py

```diff
@@ -206,13 +209,15 @@ def forward(self, images):
                 c_p = c_r[idx]  # NxP
                 e_p = e_r[idx]  # NxC
                 # decoder returns the transformed variable z and the log Jacobian determinant
-                p_u, log_jac_det = decoder(e_p, [c_p])
+                with torch.no_grad():
+                    p_u, log_jac_det = decoder(e_p, [c_p])
```

Comment:
I thought, only encoder model is fixed. Wouldn't this fix the decoder too? how is the decoder params learn then? Am I missing the point?

Reply:
The `training_step` contains a lot of similar code which trains the decoder. The only difference is that the forward function is used for inference. The original code had a few detach statements which broke the onnx graph. I just replaced the detach with `no_grad`.

Reply:
ah thanks for the clarification.